.github/workflows/build.yml

```diff
@@ -0,0 +1,95 @@
+name: Build
+
+on:
+    push:
+        branches: [master]
+    pull_request:
+    release:
+        types: [created]
+
+jobs:
+    tests:
+        runs-on: ubuntu-latest
+        name: Build and test
+        strategy:
+            fail-fast: false
+            matrix:
+                include:
+                    -   php: 7.2
+                        publish-phar: true
+                    -   php: 7.2
+                        skip-interop: true
+                    -   php: 7.2
+                        composer-flags: --prefer-lowest
+                    -   php: 7.3
+                    -   php: 7.4
+                    -   php: 7.4
+                        symfony-version: ^4.4
+                    -   php: 8.0
+
+        steps:
+            -   uses: actions/checkout@v2
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    ini-values: "phar.readonly=0,zend.exception_ignore_args=Off"
+                    coverage: none
+
+            -   name: Update Symfony version
+                if: matrix.symfony-version != ''
+                run: composer require --no-update "symfony/symfony:${{ matrix.symfony-version }}"
+
+            -   name: Uninstall container interop
+                if: matrix.skip-interop
+                run: composer remove --dev --no-update container-interop/container-interop
+
+            -   name: Install dependencies
+                run: composer update ${{ matrix.composer-flags }}
+
+            -   name: Run tests (phpunit)
+                run: ./vendor/bin/phpunit
+
+            -   name: Run tests (Behat)
+                run: ./bin/behat -fprogress --strict $([ -z "${{ matrix.skip-interop }}" ] || echo "--tags ~@interop&&~@php8")
+
+            -   name: Run tests (Behat for PHP 8.0)
+                if: matrix.php >= 8.0
+                run: ./bin/behat -fprogress --strict --tags=@php8
+
+            -   name: Build the PHAR
+                if: matrix.php != '8.0'
+                run: |
+                    curl -LSs https://box-project.github.io/box2/installer.php | php &&
```

Comment:
This downloads a very old version of box. As far as I understand that script it downloads one of these promoted files https://box-project.github.io/box2/manifest.json.

Reply:
Yes, my approach is to migrate everything as is and do any updates as a separate step.

Behat still uses box1. I think @ciaranmcnulty would appreciate if someone upgraded it and added phar scoping ;)

Reply:
I don't think we can scope phars: behat extensions have to deal with Symfony components, not only with the Behat code itself.

Reply:
This only means we can't scope PHARs fully (particularly in places where the contract of a Symfony component is exposed to the end-users).

Reply:
if we don't scope Behat packages and we don't scope Symfony components, there isn't much left to scope: https://github.com/Behat/Behat/blob/304661d4bc00934314c835c79fe36c0ce38da268/composer.json#L16-L28

Reply:
Hm, I'd seen phar scoping as a good reason for dropping older symfony versions